### DISCUSSION TITLE
Adding waves to progress bar

### DISCUSSION THREAD

**Discussion by Bellavene** [tomasklaen/uosc#741] [Ideas]
Can we use [ffmpeg waveform](https://trac.ffmpeg.org/wiki/Waveform) to add a waveform to the progress bar?

**Comment by tomasklaen** [comment]
Unlikely. We can't draw an image and than draw stuff on top of it, so it'd cover the timeline. Also from my experience with ffmpeg waveform, it takes a lot of time to render these on longer videos, and some audio formats make it even worse. There's also no way to get "progress" steps where ffmpeg would output what's currently rendered every second, so you're just waiting there for a long time while your CPU fan is trying to fly off and after a long time it'd pop in. Just not a good experience.

And I'm not sure if we can even resize & stretch the image with mpv's overlay API, which might mean we'd have to re-render the waveform every time window changes.

Another alternative would be to make our own binary to generate waveform data, and use libass to render it as a bunch of couple pixels wide rectangles, but I guarantee that would hit libass limits and make uosc start stuttering. And I also don't have time to develop this.

**Reply by Bellavene** [reply]
Understandable. I believe there must be a way to render them fast, but I got your point. Thank you.

**Reply by Bellavene** [reply]
But for clarification, I mean not dynamical waveforms, but static for the length of the clip. So it should not hit rendering times hard enough, I believe.

**Comment by Bellavene** [comment]
Should be pretty easy, it draws instantly `mpv --lavfi-complex="[aid1] asplit [ao] [v]; [v]showwavespic=s=1024x320:colors=#545454, format=rgb0 [vo]" name.mp3` I don't know is it so hard to export it and add as overlay to the progress bar?
